packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1107,6 +1090,86 @@ export class TransactionController extends BaseController<
     return this.nonceTracker.getNonceLock(address);
   }
 
+  private onBootCleanup() {
+    this.initApprovalsOfUnapprovedTransactions();
+    this.loadGasValuesOfUnapprovedTransactions();
+    this.submitApprovedTransactions();
+  }
+
+  /**
+   * Create approvals for all unapproved transactions on current chain.
+   */
+  private initApprovalsOfUnapprovedTransactions() {
+    const chainId = this.getChainId();
+    const unapprovedTransactions = this.state.transactions.filter(
+      (transaction) =>
+        transaction.status === TransactionStatus.unapproved &&
+        transaction.chainId === chainId,
+    );
+
+    for (const transactionMeta of unapprovedTransactions) {
+      this.processApproval(transactionMeta, {
+        shouldShowRequest: false,
+      }).catch((error) => {
+        /* istanbul ignore next */
+        console.error('Error during persisted transaction approval', error);
```

Comment:
Probably we will see some conflicts because https://github.com/MetaMask/core/pull/1867 or we could unify the PRs if makes sense. 
The only important point is to include the skip in case the user rejects:
```
        if (error?.code === errorCodes.provider.userRejectedRequest) {
          return;
        }
        console.error('Error during persisted transaction approval', error);
```

Reply:
Yours is approved, you go first and I will include it in mine, thanks a lot for pointing it out 👍